src/reducers/adminReducer.js

```diff
@@ -46,87 +43,62 @@ export default function admin(state = initialState.Admin, { payload, type, }) {
 
         default:
-            return state;   
+            return state;
     }
 }
 
-const updatePriceRule = (state, payload, action) => {
-    let {response} = payload;
-    let {data} = response;
-    let {PriceRules} = state;
-    switch(action){
-        case "GET":{
-            PriceRules = data;
+const updatePriceRule = (state, {response}, action) => {
+    const {data} = response;
+    let newPriceRules = [];
+    const {PriceRules} = state;
+    switch (action) {
+        case "GET":
+            newPriceRules = data;
             break;
-        }
-        case "POST":{
-            PriceRules.push(data);
+        case "POST":
+            newPriceRules = [...PriceRules, data];
             break;
-        }
-        case "PATCH":{
-            let updatedPriceRule = PriceRules.find((rule) => {return rule.id === data.id});
-            PriceRules = PriceRules.map((rule)=>{
-                if(rule.id === data.id){
-                    return updatedPriceRule;
-                } else {
-                    return rule;
-                }
-            });
-        }
+        case "PATCH":
+            newPriceRules = PriceRules.map((rule) =>
+                rule.id === data.id ? data : rule);
+        // no default
     }
     return {
         ...state,
-        PriceRules,
-    }
+        "PriceRules": newPriceRules,
+    };
 };
 
-const updateUnpaid = (state, payload, action) => {
-    let {response} = payload
-    let {data} = response;
-
-    switch(action) {
-        case "GET":{
-            
-            return {
-                ...state,
-                Unpaid:data,
-            };
-    }
-}
-}
+const handleUnpaidFetch = (state, {response}) => ({
+    ...state,
+    "Unpaid": response.data,
+});
 
-const updateDiscount = (state, payload, action, discountType) => {
-    let {response, id} = payload;
-    let {data} = response;
-    let {Discounts} = state;
-    switch(action){
-        case "GET":{
-            Discounts[discountType] = data;
+const updateDiscount = (state, {response, id}, action, discountType) => {
+    const {data} = response;
+    const {Discounts} = state;
+    let newDiscounts = [];
+    switch (action) {
+        case "GET":
+            newDiscounts = data;
             break;
-        }
-        case "POST":{
-            Discounts[discountType].push(data);
+        case "POST":
+            newDiscounts = [...Discounts[discountType], data];
             break;
-        }
-        case "PATCH":{
-            let updatedDiscount = data;
-            Discounts[discountType] = Discounts[discountType].map((discount)=>{
-                if(discount.id === id){
-                    return updatedDiscount;
-                } else {
-                    return discount;
-                }
-            });
+        case "PATCH":
+            newDiscounts = Discounts[discountType].map((discount) =>
+                discount.id === id ? data : discount);
```

Comment:
lol. i'm also dumb. i realized this was the dumb way to update an item in an array here. We should be finding the index of the discount/item to update, then only updating that specific one instead of going through the entire array. (This would be O(const) instead of O(n) I think).